streamz/sources.py

```diff
@@ -503,9 +505,11 @@ def checkpoint_emit(_part):
                         continue
                     current_position = self.positions[partition]
                     lowest = max(current_position, low)
+                    if high > lowest + self.max_batch_size:
+                        high = lowest + self.max_batch_size
```

Comment:
This is not tested, right?

Reply:
Correct, I am just writing up a test right now.

Reply:
Done